### PR TITLE
Js use armor

### PR DESCRIPTION
**Pull request by jstebel**
Remove template parameter "dim" from Quadrature.

### CODE REVIEW

File: src/quadrature/quadrature.hh [hunk 1]
```diff
@@ -45,60 +45,90 @@
...
 
     /// Returns number of quadrature points.
-    const unsigned int size() const;
+    inline const unsigned int size() const
+    { return weights.size(); }
 
     /// Returns the <tt>i</tt>th quadrature point.
-    arma::vec::fixed<dim> point(const unsigned int i) const;
+    template<unsigned int point_dim>
+    inline arma::vec::fixed<point_dim> point(const unsigned int i) const
```
Comment (jbrezmorf) [inline src/quadrature/quadrature.hh]: We can simplify the interface by returning directly the Mat object returned by get<point_dim>(i).
This way we can assign to the point and we can drop the set_point method. Same for the weights.

File: src/quadrature/quadrature.cc [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+{}
+
+
+
+
+
+
+
+template<unsigned int quad_dim>
+Quadrature quadrature_from_side(const Quadrature &subq, unsigned int sid, unsigned int pid)
```
Comment (jbrezmorf) [inline src/quadrature/quadrature.cc]: I suppose this is a replacement for the constructor? However as the diff takes it as a completely new code it is hard to see the changes. Better discuss it personally.

File: src/system/armor.hh [hunk 3]
```diff
@@ -162,7 +162,7 @@ public:
     inline Mat<Type,nr,nc> get(uint i) const
     {
         ASSERT_DBG( (nr == nRows) && (nc == nCols) );
-        return Mat<Type,nr,nc>( const_cast<Type*>(data.data()) + i*nRows*nCols );
+        return Mat<Type,nr,nc>( (Type *)data.data() + i*nRows*nCols );
```
Comment (jbrezmorf) [inline src/system/armor.hh]: We will probably need both const and non-const get method to allow assignenment.

**Comment by jbrezmorf** [comment]
Great. Just few points to discuss, possibly on Tuesday.

File: src/quadrature/quadrature.cc (hunk 2 above)
Comment (jstebel) [reply src/quadrature/quadrature.cc]: Yes, this method was originally constructor. I was not able to declare a templated constructor, hence moved it to a templated function.

File: src/quadrature/quadrature.hh (hunk 1 above)
Comment (jstebel) [reply src/quadrature/quadrature.hh]: Fixed.

File: src/system/armor.hh (hunk 3 above)
Comment (jstebel) [reply src/system/armor.hh]: This method already allows assignment. Do we need some other version?

**Pull request merged**